apps/preview/next/pages/showcases/Gallery/GalleryWithBullets.tsx

```diff
@@ -0,0 +1,100 @@
+/* eslint-disable react/no-array-index-key */
+/* eslint-disable jsx-a11y/anchor-has-content */
+import { ShowcasePageLayout } from '../../showcases';
+
+// #region source
+import { type KeyboardEvent, useState } from 'react';
+import { SfScrollable, SfButton, SfIconChevronLeft, SfIconChevronRight } from '@storefront-ui/react';
+import * as Images from './images';
+
+const images = [
+  { imageSrc: Images.gallery1Src, alt: 'backpack' },
+  { imageSrc: Images.gallery2Src, alt: 'backpack' },
+  { imageSrc: Images.gallery3Src, alt: 'backpack' },
+  { imageSrc: Images.gallery4Src, alt: 'backpack' },
+  { imageSrc: Images.gallery5Src, alt: 'backpack' },
+  { imageSrc: Images.gallery6Src, alt: 'backpack' },
+  { imageSrc: Images.gallery7Src, alt: 'backpack' },
+  { imageSrc: Images.gallery8Src, alt: 'backpack' },
+  { imageSrc: Images.gallery9Src, alt: 'backpack' },
+  { imageSrc: Images.gallery10Src, alt: 'backpack' },
+];
+
+export default function GalleryWithBullets() {
+  const [activeIndex, setActiveIndex] = useState(0);
+  const keyDownHandler = (e: KeyboardEvent<HTMLButtonElement>, index: number) => {
+    if (e.key === 'Tab') setActiveIndex(index + 1);
+    if (e.key === 'Tab' && e.shiftKey) setActiveIndex(index - 1);
+  };
```

Comment:
If I get it right, you're trying to intercept focus change and its direction to change active item accordingly. However, if you do it this way, then it won't work for example when you set focus using `element.focus()` API. Also, I'm not sure, but I'm guessing assistive technologies won't emit key event when changing focus, so it won't change as well. Another thing that bothers me a bit is the fact that the buttons you use don't have onClick functionality while they are treated as focusable elements which may be confusing.

In this case, I'm thinking if changing image onFocus is in fact a good idea? Maybe we should let the user just click on the bars? It'd let to switch to the particular image and also works more as expected, so focus, then click. What do you think?

Reply:
There was an assumption from the designers that it should not be clickable (as the elements are small) so I tried to make buttons changing the images on key press. I don't think, it should change on focus, but the focus should change after key press. Is this the wrong approach?

Reply:
Finally, after consulting with the designer, I changed the buttons to clickable.